server/models/dataset_helpers/image_metadata.py

```diff
@@ -737,15 +741,31 @@ def _populateMetadata(acquisition, clinical):
 
     # Set melanocytic field based on diagnosis
     if diagnosis in MelanocyticFieldParser.melanocyticDiagnoses:
-        if clinical.get('melanocytic') is None:
+        melanocytic = clinical.get('melanocytic')
+        if melanocytic is None:
+            clinical['melanocytic'] = True
+        elif not melanocytic:
             clinical['melanocytic'] = True
+            warnings.append('corrected inconsistent value for field %r based on field %r '
+                            '(new value: %r, %r: %r)' %
+                            (MelanocyticFieldParser.name, DiagnosisFieldParser.name,
```

Comment:
As these template strings get more complicated, it might be helpful to switch to passing a `dict` as the injected values. Then you can refer to values as `%(keyName)s` instead of just `%s`.

Reply:
Since that requires specifying dictionary keys, this seems less verbose. But I'll consider that in the future.

Reply:
Yeah, it's a trade-off, for slightly more typing, in exchange for less reliance of magical positioning of variables within the template string.